web/src/main/java/org/wso2/testgrid/web/sso/SSOSessionCheckFilter.java

```diff
@@ -55,38 +56,36 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
                          FilterChain filterChain) throws IOException, ServletException {
         //Skip checking for session if disabled in property file.
         String isSsoEnabled = ConfigurationContext.getProperty(ConfigurationProperties.ENABLE_SSO);
+        String ssoLoginUrl = ConfigurationContext.getProperty(ConfigurationProperties.SSO_LOGIN_URL);
+        UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
         if (isSsoEnabled != null) {
-            if (ConfigurationContext.getProperty(ConfigurationProperties.ENABLE_SSO).equals("false")) {
-                filterChain.doFilter(servletRequest, servletResponse);
-            }
-        } else {
-            logger.error(StringUtil.concatStrings(ConfigurationProperties.ENABLE_SSO.toString(),
-                    " is not set in ", TestGridConstants.TESTGRID_CONFIG_FILE));
-        }
-
-        String path = ((HttpServletRequest) servletRequest).getRequestURI();
-        if (isSecuredAPI(path)) {
-            Boolean isSessionValid = ((HttpServletRequest) servletRequest).isRequestedSessionIdValid();
-            if (!isSessionValid) {
-                HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
-                String ssoLoginUrl = ConfigurationContext.getProperty(ConfigurationProperties.SSO_LOGIN_URL);
-                if (ssoLoginUrl != null) {
-                    //If the request is for a backend API, Status Code 401 is sent.
-                    if (path.startsWith(Constants.BACKEND_API_URI)) {
-                        httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED);
-                    } else {
-                        httpResponse.sendRedirect(ssoLoginUrl);
-                    }
-                } else {
+            if (Boolean.valueOf(isSsoEnabled)) {
+                if (ssoLoginUrl == null || !urlValidator.isValid(ssoLoginUrl)) {
                     throw new ServletException(
-                    StringUtil.concatStrings(ConfigurationProperties.SSO_LOGIN_URL.toString(),
-                            " is not set in ", TestGridConstants.TESTGRID_CONFIG_FILE));
+                            StringUtil.concatStrings(ConfigurationProperties.SSO_LOGIN_URL.toString(),
```

Comment:
Added changes f0915eedc0725b89f92ada5b11ab8a0099065525. 
Before executing this line its check whether ssoLoginUrl is null or not. If ssoLoginUrl is null message won't be meaningful. Hence instead of replacing the ConfigurationProperties.SSO_LOGIN_URL.toString() with ssoLoginUrl, added meaningful error message.